### PR TITLE
DYN-6364 Skip Lucene indexing process in UIless mode

### PR DESCRIPTION
**Pull request by QilongTang**
### Purpose

Skip the node indexing process when DynamoModel launches in UIless mode (usually happens in Player context), so there will be no index files created or locked. I believe this one line fix will work because Lucene Initialization, e.g. indexwriter creation is part of DynamoModel initialization. As a result, as long as indexwriter is not created, any indexing process later will not work and will not create any index and lock them or overwrite the existing index.

Notice this is a quick fix for 2.19.4, for future 2.19.x hotfixes, we may want to adopt the Lucene index improvements from master branch.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Dynamo and Alias Dynamo Player can be launched at the same time

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Review by reddyashish — APPROVED** [review]

**Comment by reddyashish** [comment]
Did you get a chance to test this in Revit?

**Review by zeusongit — APPROVED** [review]

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs [hunk 1]
```diff
@@ -60,6 +60,8 @@ internal LuceneSearchUtility(DynamoModel model)
         /// </summary>
         internal void InitializeLuceneConfig(string dirName, LuceneStorage storageType = LuceneStorage.FILE_SYSTEM)
         {
+            if (DynamoModel.IsHeadless) return;
```
Comment (BogdanZavu) [inline src/DynamoCore/Utilities/LuceneSearchUtility.cs:63]: So this should be set from all CLIs and D4R when the model starts without UI right ? 
Also we should check with D4C3D.

**Comment by QilongTang** [comment]
> Did you get a chance to test this in Revit?

The bug is not reproducible in Revit because Revit Dynamo Player always try to reuse the same DynamoModel when launching Dynamo, in that case, it will be the same indexwriter so search works there. In Alias case, the Alias Dynamo Player and Dynamo are launched as two different processes,

**Review by BogdanZavu — APPROVED** [review]

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs (hunk 1 above)
Comment (QilongTang) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs:63]: I think so

**Comment by QilongTang** [comment]
@reddyashish Also I think this PR will replace https://github.com/DynamoDS/Dynamo/pull/14440 so we do not have to touch the startConfig, what do you think?

**Comment by reddyashish** [comment]
yeah wouldn't need it. Closed that PR

File: src/DynamoCore/Utilities/LuceneSearchUtility.cs (hunk 1 above)
Comment (mjkkirschner) [inline src/DynamoCore/Utilities/LuceneSearchUtility.cs:63]: perhaps you should mention this change in the comments for IsHeadLess mode, or do you not intend to port this change to master?

Comment (QilongTang) [reply src/DynamoCore/Utilities/LuceneSearchUtility.cs:63]: I can update comments there

**Comment by QilongTang** [comment]
The reported regression `DynamoCoreWpfTests.CoreUserInterfaceTests.WorkspaceContextMenu_TestIfOpenOnRightClick` is sporadic and passed locally.

**Pull request merged**